storefront/common/components/ProductImageGallery.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
+import { Figure } from 'react-bootstrap';
+
+export interface IProductImageGalleryProps {
+  listImages: string[];
+}
+
+export function ProductImageGallery({ listImages }: IProductImageGalleryProps) {
+  const MAX_IMAGES = 3;
+
+  const [currentShowUrl, setCurrentShowUrl] = useState<string>(listImages[0]);
+  const [currentIndex, setCurrentIndex] = useState<number>(0);
+  const [startSliderIndex, setStartSliderIndex] = useState(0);
```

Comment:
Hi @khoahd7621 , can we use only one State ?
const [currentIndex, setCurrentIndex] = useState<number>(0);

I think currentIndex and listImages are enough to handle all events

Reply:
Thanks for your feedback. I tried to optimize and was done. Please review again, and don't hesitate to give me feedback.